src/com/massivecraft/factions/cmd/CmdFactionsPowerBoost.java

```diff
@@ -42,46 +46,79 @@ public CmdFactionsPowerBoost()
 	// -------------------------------------------- //
 	// OVERRIDE
 	// -------------------------------------------- //
-	
-	@Override
-	public void perform() throws MassiveException
+
+	@Override public void perform() throws MassiveException
 	{
-		String type = this.<String>readArg().toLowerCase();
+		String receiverType = this.<String>argAt(0).toLowerCase();
 		boolean doPlayer = true;
-		if (type.equals("f") || type.equals("faction"))
+		if (receiverType.equals("f") || receiverType.equals("faction"))
 		{
 			doPlayer = false;
 		}
-		else if (!type.equals("p") && !type.equals("player"))
+		else if (!receiverType.equals("p") && !receiverType.equals("player"))
 		{
 			msg("<b>You must specify \"p\" or \"player\" to target a player or \"f\" or \"faction\" to target a faction.");
-			msg("<b>ex. /f powerboost p SomePlayer 0.5  -or-  /f powerboost f SomeFaction -5");
+			msg("<b>ex. /f powerboost p SomePlayer a 1  -or-  /f powerboost f SomeFaction r 5 -or- /f powerboost f SomePlayer s 50");
 			return;
 		}
-		
-		double targetPower = this.readArgAt(2);
 
-		String target;
+		GiveType giveType = getGiveType(argAt(2));
+		double targetPower = this.readArgAt(3);
 
+		if (giveType == null)
+		{
+			msg("<b>You must specify \"a\" or \"add\" to add, \"r\" or \"remove\" to remove, or \"s\" or \"set\" to set the powerboost of a player/faction");
+			msg("<b>ex. /f powerboost p SomePlayer a 1  -or-  /f powerboost f SomeFaction r 5 -or- /f powerboost f SomePlayer s 50");
+			return;
+		}
+
+		String target;
 		if (doPlayer)
 		{
-			this.getParameters().set(1, parameterMplayer);
 			MPlayer targetPlayer = this.readArgAt(1);
-			
-			targetPlayer.setPowerBoost(targetPower);
-			target = "Player \""+targetPlayer.getName()+"\"";
+
+			if (giveType == GiveType.ADD || giveType == GiveType.REMOVE) targetPlayer.addPowerBoost(targetPower);
```

Comment:
These lines ache me, because they repeat just a few lines down with the only difference being a faction instead of a player.

My suggestion: Faction and MPlayer should both implement an interface called Powerboostable (WIP name) that offer the methods addPowerBoost(Double amount), getPowerBoost() and setPowerBoost(Double amount).

This way, you could almost merge the if statement.
To get the name you can cast the object to Named like this.

String targetName = ((Named) targetBoostable).getName();